src/rstar.jl

```diff
@@ -2,14 +2,16 @@
     rstar(
         rng=Random.GLOBAL_RNG,
         classifier,
-        samples::AbstractMatrix,
-        chain_indices::AbstractVector{Int};
+        samples::AbstractArray,
+        [chain_indices::AbstractVector{Int}];
```

Comment:
This seems a bit confusing: According to the docstring we only support matrices and 3-dim arrays not generic `AbstractArray`, and it seems `chain_indices` is not supported if `samples` is a 3-d array. Maybe the only clear documentation are separate docstrings (where in the text one could refer to the other)?